libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2295,13 +2295,13 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 			}
 			case DataLocation::Memory:
```

Comment:
What about the next case, `DataLocation::CallData`? It seems to use the same process of concatenating strings resulting from a function call.

Reply:
It's fine. It avoids this problem by calculating `indexAccessFunction` before the concatenation. The problem was caused by two accesses to `m_utils` with undefined order (one explicitly and one hidden inside `expressionAsType()`). In the `CallData` case the order is already well defined.

I could have actually formulated the code in `Memory` case the same way `CallData` does it, but I thought that creating more local variables made this more readable. Maybe I should change the `CallData` case to match.

Reply:
I refactored to calldata part to look like the one for memory.